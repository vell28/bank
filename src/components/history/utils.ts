import { isToday, isYesterday, format } from 'date-fns';
import { pathOr } from 'ramda';

import { ITransactionList, ITransaction, IOrderedTransactions } from 'models/transactions/entities';

export const getDayTitle = (date: Date): string => {
  if (isToday(date)) {
    return 'today';
  }
  if (isYesterday(date)) {
    return 'yesterday';
  }
  return format(date, 'dd/MM/yyyy');
};

export const groupByAccountType = (list: ITransactionList): IOrderedTransactions => {
  const ordered: IOrderedTransactions = {};
  list.map((item: ITransaction) => {
    const type = pathOr('unknown', ['simpleAccount', 'accountType'])(item);
    if (ordered[type]) {
      ordered[type].push(item);
    } else {
      ordered[type] = [item];
    }
    return item;
  });
  return ordered;
};
