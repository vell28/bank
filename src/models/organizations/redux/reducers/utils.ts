import {
  findIndex, pathOr, pipe, propEq
} from 'ramda';

import { IAddress } from 'models/operations/order-card/entities';
import { IOrganization } from 'models/organizations/entities';
import getDataServer from '../../../../modules/countries/country-list.json';
import { IServerAddress, IServerOrganization } from '../actions/entities.js';

export const findAccountIndexById = (id: string) =>
  pipe(pathOr([], ['data', 0, 'accounts']), findIndex(propEq('id', id)));

// alias
type countryCode = string;

const parseCountry = (country: string): countryCode => {
  const targetCountry = getDataServer.find(propEq('name', country));
  if (targetCountry) {
    return targetCountry.code;
  }
  throw new Error(`Unknown country ${country}`); // nothing else can do
};

const parseApartmentNumber = (serverRepresentation: string): number => {
  const parsed = parseInt(serverRepresentation, 10);
  // eslint-disable-next-line no-restricted-globals
  if (!isNaN(parsed)) {
    return parsed;
  }
  return 0;
  // throw new Error('Invalid apartmentNumber');
};

// inthis context client means client-side(front-end) application
const parseSingleAddressToClientType = ({
  country = '',
  province,
  city = '',
  postalCode = '',
  street = '',
  streetNumber = '',
  apartmentNumber = '0',
}: IServerAddress): IAddress => ({
  countryCode: parseCountry(country),
  province,
  city,
  postalCode,
  street,
  streetNumber,
  apartmentNumber: parseApartmentNumber(apartmentNumber),
});

export const parseOrganizationsToClientType = (organizations: IServerOrganization[]): IOrganization[] => {
  return organizations.map((item) => {
    return {
      ...item,
      client: {
        ...item.client,
        // we expect server to send us an address
        address: parseSingleAddressToClientType(item.client.address as IServerAddress),
      },
    };
  });
};
