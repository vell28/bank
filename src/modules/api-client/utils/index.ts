import {
  propSatisfies,
  ifElse,
  is,
  and,
  lte,
  gte,
  always,
  propEq,
  equals,
  either,
  pipe,
  head,
  propOr,
  pathOr,
  pathEq,
} from 'ramda';
import { stringify } from 'qs';

import {
  IError, IFetchResponse, IOptions, IRequestParams, ResponseAPI
} from '../entities';
import { appErrorStatuses, METHODS } from './const';

const inInterval = (start: number, stop: number) =>
  ifElse(is(Number), (status: number) => and(gte(status, start), lte(status, stop)), always(false));

export const is20x = propSatisfies(inInterval(200, 299), 'status');
export const is40x = propSatisfies(inInterval(400, 499), 'status');
export const is50x = propSatisfies(inInterval(500, 599), 'status');

export const isNetworkError = propEq('status', -1);
export const isGET = equals(METHODS.GET);

export const getConfirmFromResponseToken = (error: IError): string =>
  pipe(propOr([], 'errors'), head, pathOr('', ['properties', 'confirmationToken']))(error);

export const getConfirmCodeLengthToken = (error: IError): number =>
  pipe(propOr([], 'errors'), head, pathOr(0, ['properties', 'confirmationCodeLength']))(error);

export const getHTTPStatusCode = (error: IError): number =>
  pipe(
    propOr([], 'errors'),
    head,
    // propOr(0, 'code')
    // red error unknown not assignable to number
    // even seem to be right overload propOr<number>
    pathOr(0, ['code']),
  )(error);

export const isAllowBodyMethod = either(equals(METHODS.POST), equals(METHODS.PATCH));

export const onUnknow = (response: IFetchResponse): Promise<IError> =>
  // eslint-disable-next-line prefer-promise-reject-errors
  Promise.reject({
    ok: false,
    status: appErrorStatuses.UNKNOWN.code,
    errors: [
      {
        code: appErrorStatuses.UNKNOWN.code,
        type: response.statusText,
        message: 'Unknown fetch error.',
      },
    ],
  });

export const makeUrl = (method: string, url: string, data?: any): string =>
  isGET(method) && data ? `${url}${stringify(data, { addQueryPrefix: true })}` : url;

export const makeRequestOptions = (
  method: string = METHODS.GET,
  { data, headers, isFormData }: IRequestParams,
): IOptions => {
  const options = {
    method,
    headers,
  };

  if (isAllowBodyMethod(method) && data) {
    let body: any;
    if (isFormData) {
      body = new FormData();
      // eslint-disable-next-line no-restricted-syntax
      for (const k in data) {
        // eslint-disable-next-line no-prototype-builtins
        if (data.hasOwnProperty(k)) {
          body.append(k, data[k]);
        }
      }
      body = new URLSearchParams(body);
    } else {
      body = JSON.stringify(data);
    }
    return {
      ...options,
      body,
    };
  }

  return options;
};

export const isSuccessResponse = (resp: ResponseAPI<any>): boolean => pathEq(['data', 'status'], 'ok')(resp);
