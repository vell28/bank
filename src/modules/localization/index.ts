// @flow

import i18n from 'i18next';
import { find, test } from 'ramda';
import { initReactI18next } from 'react-i18next';
import LanguageDetector from 'i18next-browser-languagedetector';

// ToDo: need sync to existed locales
import { de, en } from './locales';
import { ILanguage, LangKeysType, LanguageKeys } from './entities';

export const languagesList: ILanguage[] = [
  { key: LanguageKeys.EN, nativeName: 'English' },
  { key: LanguageKeys.DE, nativeName: 'Deutsch' },
];

export const languagesMap = {
  [LanguageKeys.DE]: 'Deutsch',
  [LanguageKeys.EN]: 'English',
  [LanguageKeys.RU]: 'Russian',
};

export const getActiveLanguageKey = (activeLanguage: string = i18n.language): LangKeysType => {
  // When localStorage is empty, activeLanguage='en-GB'
  const item = find((lang: ILanguage) => test(new RegExp(lang.key, 'gi'), activeLanguage), languagesList);
  if (item) {
    return item.key;
  }
  return LanguageKeys.EN;
};

export const changeLanguage = (lng: LangKeysType) => {
  window.location.reload();
  i18n.changeLanguage(lng);
};

/**
 * Detector Options
 * https://github.com/i18next/i18next-browser-languageDetector#detector-options
 */
const detection = {
  caches: ['localStorage'],
};

/**
 * ISO 639
 * https://www.iso.org/iso-639-language-codes.html
 * http://www.loc.gov/standards/iso639-2/php/code_list.php
 */
i18n
  // .use(Locize)
  .use(initReactI18next) // passes i18n down to react-i18next
  .use(LanguageDetector)
  .init({
    ns: ['common'],
    defaultNS: 'common',

    resources: { en, de },

    fallbackLng: LanguageKeys.EN,

    nsSeparator: '::',
    keySeparator: false,

    interpolation: {
      escapeValue: false, // not needed for react as it escapes by default
    },

    detection,
  });

export default i18n;
