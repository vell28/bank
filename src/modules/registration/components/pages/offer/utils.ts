import { IRegistrationOfferProps, IRegistrationOfferNavigationProps } from './types';

export const selectNavigationalProps = ({
  onDataProtectionPolicyClick,
  onGeneralTermsAndConditionsClick,
  onPreContractalInformationClick,
  onPriceListClick,
  onSecureUsageOfTheCardClick,
}: IRegistrationOfferProps): IRegistrationOfferNavigationProps => {
  return {
    onDataProtectionPolicyClick,
    onGeneralTermsAndConditionsClick,
    onPreContractalInformationClick,
    onPriceListClick,
    onSecureUsageOfTheCardClick,
  };
};
