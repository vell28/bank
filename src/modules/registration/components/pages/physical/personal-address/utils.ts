import { IPersonalAddressData, IAddressData } from '../../../../models/pages/physical/personal-address/entities';

export const getAddress = (addressDeliver: IPersonalAddressData | IAddressData) => {
  const {
    street, houseNumber, city, postalCode
  } = addressDeliver;
  const newStreet = street || houseNumber ? `${street} ${houseNumber}, ` : '';
  const newCity = city ? `${city}, ` : '';

  return {
    address: newStreet || newCity || postalCode ? `${newStreet}${newCity}${postalCode}` : '',
    isFullAddress: street && houseNumber && city && postalCode,
  };
};
