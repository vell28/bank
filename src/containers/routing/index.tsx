import React from 'react';

import { Route, Switch } from 'react-router-dom';

import {
  BonusesPage,
  InsurancePage,
  InvestmentsPage,
} from '../pages';

import BusinessPage from '../../containers/pages/business-page';

import NewAccountPage from '../pages/new-account-page';
import ProfilePage from '../pages/profile-page';

import BankingRoutes from './routes/banking';
import { PATHS_CONFIG } from './utils';

const {
  banking,
  banking_account,
  new_account,
  bonuses,
  cryptotrading,
  insurance,
  business,
  profile,
} = PATHS_CONFIG;

export default () => (
  <Switch>
    <Route
      path={[banking.path, banking_account.path]}
      component={BankingRoutes}
      exact
    />
    <Route path={new_account.path} component={NewAccountPage} exact />
    <Route path={bonuses.path} component={BonusesPage} exact />
    <Route path={cryptotrading.path} component={InvestmentsPage} exact />
    <Route path={insurance.path} component={InsurancePage} exact />
    <Route path={business.path} component={BusinessPage} exact />
    <Route path={profile.path} component={ProfilePage} exact />
  </Switch>
);
