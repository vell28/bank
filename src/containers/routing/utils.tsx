import {
  findIndex, isNil, pipe, pathOr, forEach
} from 'ramda';
import { matchPath } from 'react-router';
import { RouteComponentProps } from 'react-router-dom';
import { connectedRouterRedirect } from 'redux-auth-wrapper/history4/redirect';

import { CurrencyCodeType } from 'modules/currencies';
import { getToken, getFirebaseToken } from 'models/authorization/redux/selectors';
import { IStore } from 'modules/store/types';

enum RoutePaths {
  BANKING = 'banking',
  BANKING_ACCOUNT = 'banking_account',
  NEW_ACCOUNT = 'new_account',
  BONUSES = 'bonuses',
  BUSINESS = 'business',
  INSURANCE = 'insurance',
  CRYPTOTRADING = 'cryptotrading',
  PROFILE = 'profile',
  AUTHORIZATION = 'authorization',
  SIGNIN = 'signin',
  SIGNIN_CONFIRM = 'signin_confirm',
  TOPUP_PUBLIC = 'topup_public',
}

export type RoutePathType =
  | RoutePaths.BANKING
  | RoutePaths.BANKING_ACCOUNT
  | RoutePaths.NEW_ACCOUNT
  | RoutePaths.BONUSES
  | RoutePaths.BUSINESS
  | RoutePaths.CRYPTOTRADING
  | RoutePaths.PROFILE
  | RoutePaths.AUTHORIZATION
  | RoutePaths.SIGNIN
  | RoutePaths.TOPUP_PUBLIC
  | RoutePaths.INSURANCE
  | RoutePaths.SIGNIN_CONFIRM;

export const BANKING_PATH = '/';
export const BANKING_ACCOUNT_PATH = '/banking/account/:id/:currency';
export const NEW_ACCOUNT_PATH = '/create-account';
export const BONUSES_PATH = '/bonuses';
export const BUSINESS_PATH = '/business';
export const INSURANCE_PATH = '/insurance';
export const CRYPTOTRADING_PATH = 'https://frontend.stage.ak-ecosystem.com/';
export const PROFILE_PATH = '/profile';
export const AUTHORIZATION_PATH = '/authorization';
export const SIGNIN_PATH = '/authorization/signin';
export const SIGNIN_CONFIRM_PATH = '/authorization/signin/confirm';
export const ERROR_PATH = '/error';
export const TOPUP_PUBLIC_PATH = '/public-topup/:id';

export interface IPath {
  path: string;
  name: RoutePathType;
}

export interface IPathConfig {
  [key: string]: IPath;
}

export const PATHS_CONFIG: IPathConfig = {
  [RoutePaths.SIGNIN]: {
    path: SIGNIN_PATH,
    name: RoutePaths.SIGNIN,
  },
  [RoutePaths.SIGNIN_CONFIRM]: {
    path: SIGNIN_CONFIRM_PATH,
    name: RoutePaths.SIGNIN_CONFIRM,
  },
  [RoutePaths.AUTHORIZATION]: {
    path: AUTHORIZATION_PATH,
    name: RoutePaths.AUTHORIZATION,
  },
  [RoutePaths.BANKING]: {
    path: BANKING_PATH,
    name: RoutePaths.BANKING,
  },
  [RoutePaths.BANKING_ACCOUNT]: {
    path: BANKING_ACCOUNT_PATH,
    name: RoutePaths.BANKING_ACCOUNT,
  },
  [RoutePaths.NEW_ACCOUNT]: {
    path: NEW_ACCOUNT_PATH,
    name: RoutePaths.NEW_ACCOUNT,
  },
  [RoutePaths.BONUSES]: {
    path: BONUSES_PATH,
    name: RoutePaths.BONUSES,
  },
  [RoutePaths.BUSINESS]: {
    path: BUSINESS_PATH,
    name: RoutePaths.BUSINESS,
  },
  [RoutePaths.INSURANCE]: {
    path: INSURANCE_PATH,
    name: RoutePaths.INSURANCE,
  },
  [RoutePaths.CRYPTOTRADING]: {
    path: CRYPTOTRADING_PATH,
    name: RoutePaths.CRYPTOTRADING,
  },
  [RoutePaths.PROFILE]: {
    path: PROFILE_PATH,
    name: RoutePaths.PROFILE,
  },
  [RoutePaths.TOPUP_PUBLIC]: {
    path: TOPUP_PUBLIC_PATH,
    name: RoutePaths.TOPUP_PUBLIC,
  },
};

export const mainMenuPaths: IPath[] = [
  PATHS_CONFIG.banking,
  PATHS_CONFIG.bonuses,
  PATHS_CONFIG.business,
  PATHS_CONFIG.cryptotrading,
  PATHS_CONFIG.insurance,
  PATHS_CONFIG.profile,
];

const pathWithParamsForMatch: string[] = [BANKING_ACCOUNT_PATH];

export const getActiveIndex = (location: string, paths: IPath[]) =>
  pipe(
    findIndex((item: IPath) => {
      const match = matchPath(location, {
        path: item.path,
        exact: true,
      });
      return !isNil(match);
    }),
    (index: number) => (index < 0 ? 0 : index),
  )(paths);

export const getBankingAccountPath = (id: string, currency: CurrencyCodeType): string =>
  `/banking/account/${id}/${currency}`;

export const getPathParamValue = (key: string, match: string, pathList: string[] = pathWithParamsForMatch): string => {
  let matchedPath = {};
  forEach((path: string) => {
    const res = matchPath(match, {
      path,
      exact: true,
      strict: false,
    });
    if (res !== null) {
      matchedPath = res;
    }
  }, pathList);

  return pathOr('', ['params', key])(matchedPath);
};

export const getRouteParamByName = (name: string, routeProps: RouteComponentProps): string =>
  pathOr('', ['match', 'params', name], routeProps);

export const userIsAuthenticated = connectedRouterRedirect({
  redirectPath: AUTHORIZATION_PATH,
  authenticatedSelector: (store: IStore) => !!getFirebaseToken(store) || !!getToken(store),
  wrapperDisplayName: 'UserIsAuthenticated',
});
