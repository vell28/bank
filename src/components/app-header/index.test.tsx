import React from 'react';

import { ThemeProvider } from 'styled-components';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { BrowserRouter } from 'react-router-dom';

import { mainDarkTheme } from 'modules/themes';
import { mainMenuPaths } from 'containers/routing/utils';

import { languagesList } from 'modules/localization';
import { AppHeader } from '.';

describe('<AppHeader /> tests', () => {
  test('should success render, should highlight active link', () => {
    const onChangeCb = jest.fn();
    const index = 2;
    const component: any = mount(
      <BrowserRouter>
        <ThemeProvider theme={mainDarkTheme}>
          <AppHeader
            languages={languagesList}
            activeIndex={index}
            onLngChange={onChangeCb}
            activeLng="ru"
            navList={mainMenuPaths}
          >
            account
          </AppHeader>
        </ThemeProvider>
      </BrowserRouter>,
    );
    const activeLinkText = component.find('a[data-active=true]').text();
    expect(activeLinkText.toLowerCase()).toEqual(mainMenuPaths[index].name);
    expect(toJson(component)).toMatchSnapshot();
  });
});
