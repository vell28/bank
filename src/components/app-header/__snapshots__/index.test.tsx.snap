// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppHeader /> tests should success render, should highlight active link 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <ThemeProvider
      theme={
        Object {
          "breakpoints": Array [
            "576px",
            "768px",
            "992px",
            "1200px",
          ],
          "colors": Object {
            "alto": "#dedede",
            "asparagus": "#77a45c",
            "black": "#000000",
            "blue": "#008cff",
            "blueLight": "#70cde4",
            "borderLight": "#cad3db",
            "dark": "#030303",
            "darkBlue": "#011f3d",
            "darkBlue2": "#132F4B",
            "elephant": "#12304b",
            "feijoa": "#93d77e",
            "gray": "#a5adb8",
            "gray10": "#cad3db",
            "gray11": "#959fac",
            "gray2": "#a7a9ac",
            "gray3": "#a1aab6",
            "gray4": "#e8e8e8",
            "gray5": "#cbcbcb",
            "gray6": "#eaeaea",
            "gray7": "#acacac",
            "gray9": "#9faab7",
            "grayHover": "#00000008",
            "grayLight": "#cdd0d5",
            "green": "#7ed321",
            "green1": "#5ba00f",
            "green2": "#4dd964",
            "green3": "#77a45c",
            "green4": "#b1de6b",
            "greenLight": "#d9f9b6",
            "light": "#d5dae0",
            "malibu": "#53b8ff",
            "navy": "#01172d",
            "orange": "#e4be79",
            "persimmon": "#ff5353",
            "purple": "#a479cb",
            "red": "#e10606",
            "red2": "#ff2e7b",
            "red3": "#ea4848",
            "rose": "#d08484",
            "slate": "#596778",
            "submarine": "#b6bcc4",
            "turquoise": "#79cbcb",
            "white": "#ffffff",
            "white2": "#dddddd",
          },
          "dimension": Object {},
          "fonts": Object {
            "faFontsPath": "/css/webfonts",
            "fontFamily": "AvenirNext, Arial",
          },
          "name": "main-dark-theme",
          "size": "default",
          "typography": Object {
            "bodyBackground": "#011f3d",
            "bodyFontColor": "#ffffff",
            "bodyFontFamily": "AvenirNext, Arial",
            "globalFontSize": "1.6rem",
            "globalLineHeight": 1.5,
            "globalWeightNormal": 400,
            "headersFontColor": "#ffffff",
            "headersFontFamily": "AvenirNext, Arial",
            "headersFontWeight": 700,
          },
        }
      }
    >
      <AppHeader
        activeIndex={2}
        activeLng="ru"
        languages={
          Array [
            Object {
              "key": "en",
              "nativeName": "English",
            },
            Object {
              "key": "de",
              "nativeName": "Deutsch",
            },
          ]
        }
        navList={
          Array [
            Object {
              "name": "banking",
              "path": "/",
            },
            Object {
              "name": "bonuses",
              "path": "/bonuses",
            },
            Object {
              "name": "business",
              "path": "/business",
            },
            Object {
              "name": "cryptotrading",
              "path": "https://frontend.stage.ak-ecosystem.com/",
            },
            Object {
              "name": "insurance",
              "path": "/insurance",
            },
            Object {
              "name": "profile",
              "path": "/profile",
            },
          ]
        }
        onLngChange={[MockFunction]}
      >
        <styled.div>
          <div
            className="sc-AxjAm fmHMYc"
          >
            <styled.div>
              <div
                className="sc-AxirZ eJNiei"
              >
                <styled.div>
                  <div
                    className="sc-AxiKw ewoZkl"
                  >
                    <Styled(Link)
                      to="/"
                    >
                      <Link
                        className="sc-Axmtr kkOJu"
                        to="/"
                      >
                        <LinkAnchor
                          className="sc-Axmtr kkOJu"
                          href="/"
                          navigate={[Function]}
                        >
                          <a
                            className="sc-Axmtr kkOJu"
                            href="/"
                            onClick={[Function]}
                          >
                            <img
                              alt="logo"
                              src="logo.svg"
                            />
                          </a>
                        </LinkAnchor>
                      </Link>
                    </Styled(Link)>
                  </div>
                </styled.div>
                <styled.nav>
                  <nav
                    className="sc-AxhCb iCmgJC"
                  >
                    <styled.ul>
                      <ul
                        className="sc-AxhUy fISUny"
                      >
                        <styled.li>
                          <li
                            className="sc-AxgMl khpreb"
                          >
                            <Styled(Link)
                              data-active={false}
                              to="/"
                            >
                              <Link
                                className="sc-AxmLO LcVuO"
                                data-active={false}
                                to="/"
                              >
                                <LinkAnchor
                                  className="sc-AxmLO LcVuO"
                                  data-active={false}
                                  href="/"
                                  navigate={[Function]}
                                >
                                  <a
                                    className="sc-AxmLO LcVuO"
                                    data-active={false}
                                    href="/"
                                    onClick={[Function]}
                                  >
                                    Banking
                                  </a>
                                </LinkAnchor>
                              </Link>
                            </Styled(Link)>
                          </li>
                        </styled.li>
                        <styled.li>
                          <li
                            className="sc-AxgMl khpreb"
                          >
                            <Styled(Link)
                              data-active={false}
                              to="/bonuses"
                            >
                              <Link
                                className="sc-AxmLO LcVuO"
                                data-active={false}
                                to="/bonuses"
                              >
                                <LinkAnchor
                                  className="sc-AxmLO LcVuO"
                                  data-active={false}
                                  href="/bonuses"
                                  navigate={[Function]}
                                >
                                  <a
                                    className="sc-AxmLO LcVuO"
                                    data-active={false}
                                    href="/bonuses"
                                    onClick={[Function]}
                                  >
                                    Bonuses
                                  </a>
                                </LinkAnchor>
                              </Link>
                            </Styled(Link)>
                          </li>
                        </styled.li>
                        <styled.li>
                          <li
                            className="sc-AxgMl khpreb"
                          >
                            <Styled(Link)
                              data-active={true}
                              to="/business"
                            >
                              <Link
                                className="sc-AxmLO hxwued"
                                data-active={true}
                                to="/business"
                              >
                                <LinkAnchor
                                  className="sc-AxmLO hxwued"
                                  data-active={true}
                                  href="/business"
                                  navigate={[Function]}
                                >
                                  <a
                                    className="sc-AxmLO hxwued"
                                    data-active={true}
                                    href="/business"
                                    onClick={[Function]}
                                  >
                                    Business
                                  </a>
                                </LinkAnchor>
                              </Link>
                            </Styled(Link)>
                          </li>
                        </styled.li>
                        <styled.li>
                          <li
                            className="sc-AxgMl khpreb"
                          >
                            <styled.a
                              href="https://frontend.stage.ak-ecosystem.com/"
                              target="_blank"
                            >
                              <a
                                className="sc-fzozJi bxiZlP"
                                href="https://frontend.stage.ak-ecosystem.com/"
                                target="_blank"
                              >
                                AK Crypto
                              </a>
                            </styled.a>
                          </li>
                        </styled.li>
                        <styled.li>
                          <li
                            className="sc-AxgMl khpreb"
                          >
                            <Styled(Link)
                              data-active={false}
                              to="/insurance"
                            >
                              <Link
                                className="sc-AxmLO LcVuO"
                                data-active={false}
                                to="/insurance"
                              >
                                <LinkAnchor
                                  className="sc-AxmLO LcVuO"
                                  data-active={false}
                                  href="/insurance"
                                  navigate={[Function]}
                                >
                                  <a
                                    className="sc-AxmLO LcVuO"
                                    data-active={false}
                                    href="/insurance"
                                    onClick={[Function]}
                                  >
                                    Insurance
                                  </a>
                                </LinkAnchor>
                              </Link>
                            </Styled(Link)>
                          </li>
                        </styled.li>
                        <styled.li>
                          <li
                            className="sc-AxgMl khpreb"
                          >
                            <Styled(Link)
                              data-active={false}
                              to="/profile"
                            >
                              <Link
                                className="sc-AxmLO LcVuO"
                                data-active={false}
                                to="/profile"
                              >
                                <LinkAnchor
                                  className="sc-AxmLO LcVuO"
                                  data-active={false}
                                  href="/profile"
                                  navigate={[Function]}
                                >
                                  <a
                                    className="sc-AxmLO LcVuO"
                                    data-active={false}
                                    href="/profile"
                                    onClick={[Function]}
                                  >
                                    Profile
                                  </a>
                                </LinkAnchor>
                              </Link>
                            </Styled(Link)>
                          </li>
                        </styled.li>
                      </ul>
                    </styled.ul>
                  </nav>
                </styled.nav>
                <styled.div>
                  <div
                    className="sc-AxheI fJlcfU"
                  >
                    <LanguageSelector
                      activeLng="ru"
                      languages={
                        Array [
                          Object {
                            "key": "en",
                            "nativeName": "English",
                          },
                          Object {
                            "key": "de",
                            "nativeName": "Deutsch",
                          },
                        ]
                      }
                      onLngChange={[MockFunction]}
                    >
                      <styled.div>
                        <div
                          className="sc-fzoLsD kFOIiu"
                        >
                          <styled.div
                            isShown={false}
                            onClick={[Function]}
                          >
                            <div
                              className="sc-fzpans gIwDBx"
                              onClick={[Function]}
                            >
                              ru
                              <i
                                className="fas fa-chevron-down"
                              />
                            </div>
                          </styled.div>
                          <styled.div
                            isShown={false}
                          >
                            <div
                              className="sc-fzplWN jHCTbU"
                            >
                              <span
                                key="en"
                                onClick={[Function]}
                              >
                                en
                              </span>
                              <span
                                key="de"
                                onClick={[Function]}
                              >
                                de
                              </span>
                            </div>
                          </styled.div>
                        </div>
                      </styled.div>
                    </LanguageSelector>
                  </div>
                </styled.div>
              </div>
            </styled.div>
          </div>
        </styled.div>
      </AppHeader>
    </ThemeProvider>
  </Router>
</BrowserRouter>
`;
